compiler/z/codegen/OMRInstruction.cpp

```diff
@@ -653,28 +653,51 @@ bool isMatchingStoreRestore(TR::Instruction *cursorLoad, TR::Instruction *cursor
    return false;
    }
 
-uint32_t getRegMaskFromRange(TR::Instruction * inst)
+uint32_t getGPRMaskFromRange(TR::Instruction * inst, TR::CodeGenerator *cg)
    {
    uint32_t regs = 0;
 
    TR::Register *lowReg = toS390RSInstruction(inst)->getFirstRegister();
    TR::Register *highReg = toS390RSInstruction(inst)->getSecondRegister();
    if (inst->getRegisterOperand(1)->getRegisterPair())
       highReg = inst->getRegisterOperand(1)->getRegisterPair()->getHighOrder();
+   uint32_t lowRegNum = toRealRegister(lowReg)->getRegisterNumber();
+   uint32_t highRegNum = toRealRegister(highReg)->getRegisterNumber();
 
-   uint32_t lowRegNum = ANYREGINDEX(toRealRegister(lowReg)->getRegisterNumber());
-   uint32_t highRegNum = ANYREGINDEX(toRealRegister(highReg)->getRegisterNumber());
+   uint32_t minRegNum = TR::RealRegister::FirstGPR;
+   uint32_t maxRegNum = TR::RealRegister::LastAssignableGPR;
+   TR_ASSERT(lowRegNum >= minRegNum && lowRegNum <= maxRegNum && highRegNum >= minRegNum && highRegNum <= maxRegNum, "Unexpected register number");
 
-   TR_ASSERT(lowRegNum >= 0 && lowRegNum <= 15 && highRegNum >= 0 && highRegNum <= 15, "Unexpected register number");
-   for (uint32_t i = lowRegNum; ; i = ((i == 15) ? 0 : i+1))  // wrap around to 0 at 15
+   // wrap around for loop terminal index to minRegNum
+   for (uint32_t i = lowRegNum; ; i = ((i == maxRegNum) ? minRegNum : i+1))
       {
-      regs = regs | (0x1 << i);
+      regs |= cg->machine()->getRealRegister(i)->getRealRegisterMask();
       if (i == highRegNum)
          break;
       }
    return regs;
    }
-   
+
+uint32_t getVRFMaskFromRange(TR::Instruction * inst, TR::CodeGenerator *cg)
+   {
+   uint32_t regs = 0;
+
+   TR::Register *lowReg = toS390VRSInstruction(inst)->getFirstRegister();
+   TR::Register *highReg = toS390VRSInstruction(inst)->getSecondRegister();
+   uint32_t lowRegNum = toRealRegister(lowReg)->getRegisterNumber();
+   uint32_t highRegNum = toRealRegister(highReg)->getRegisterNumber();
+
+   TR_ASSERT_FATAL(lowRegNum <= highRegNum && (highRegNum - lowRegNum) < 16 &&
+                   lowRegNum >= TR::RealRegister::FirstAssignableVRF && highRegNum <= TR::RealRegister::LastAssignableVRF,
+                   "Invalid vector range");
```

Comment:
I think it'll be worthwhile updating the TR_ASSERT_FATAL message to include the values for `lowRegNum` and `highRegNum`.